### PR TITLE
feat(rest): v6 docs for REST API

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes:
v6 documentation for REST API

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by stocaaro — COMMENTED** [review]
Should this change include migrating the existing v5 content over to lib-v1 to ensure that remains available on the legacy tab?

File: src/fragments/lib/restapi/js/authz.mdx [hunk 1]
```diff
@@ -1,96 +1,122 @@
...
-        name: 'sampleCloudApi',
-        endpoint: 'https://xyz.execute-api.us-east-1.amazonaws.com/Development',
-        custom_header: async () => {
-          return { Authorization: 'token' };
-          // Alternatively, with Cognito User Pools use this:
-          // return { Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}` }
-          // return { Authorization: `Bearer ${(await Auth.currentSession()).getIdToken().getJwtToken()}` }
-        }
+    REST: {
+      headers: async () => {
```
Comment (AllanZhengYP) [inline src/fragments/lib/restapi/js/authz.mdx:17]: Right now we don't have a easy way to retrieve the current options when specifying addition options. I will create a dev task for it.

File: src/fragments/lib/restapi/js/getting-started.mdx
Comment (AllanZhengYP) [inline src/fragments/lib/restapi/js/getting-started.mdx:1]: This doc page is not deleted, but replaced by `getting-started` folder

**Comment by AllanZhengYP** [comment]
Preview branch: https://aws-amplify-docs-git-amplifyv6branchrestapi-allanzhengyp.vercel.app/lib/restapi/getting-started/

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Pull request merged**